src/pure/List.mo

```diff
@@ -97,7 +97,7 @@ module {
   /// Space: O(1)
   public func get<T>(list : List<T>, n : Nat) : ?T = switch list {
     case null null;
-    case (?(h, t)) if (n == 0) ?h else get(t, n - 1)
+    case (?(h, t)) if (n == 0) ?h else get(t, n - 1 : Nat)
```

Comment:
(FYA Another way to avoid these warning would be to add the type parameter in recursive call: e.g. get<T>(..))